dyn_em/start_em.F

```diff
@@ -992,6 +992,8 @@ SUBROUTINE start_domain_em ( grid, allowed_to_read &
                       grid%w0avg, grid%rainnc, grid%rainc, grid%raincv, grid%rainncv,  &
                       grid%snownc, grid%snowncv, grid%graupelnc, grid%graupelncv,  &
                       z_at_q, grid%alt, grid%qnwfa2d, scalar(ims,kms,jms,1), num_scalar,         & ! G. Thompson
+                      moist(ims,kms,jms,1), num_moist,                  & ! G. Thompson
```

Comment:
@gthompsnWRF 
Greg,
Instead of `moist(ims,kms,jms,1)`, why not just pass in `moist`?
I do not see a reason to do otherwise.

Reply:
This is the first time we have needed to pass the 4d moist array into phy_init, so I am wondering about that. It seems to be a big step.

Reply:
@davegill  yes, that is correct as well. I just followed the same as the scalar argument, so would you prefer a match or both changed?

@dudhia  while the question you ask about g_ratio is true, how would you suggest a different way?  Both Hugh Morrison and I use this almost identically.  The difference with gamma shape parameter as compared to using 2.5 microns drop size is dramatic. Changing "mu" in gamma distribution would be rather small difference compared to the setting of 2.5 microns. Otherwise, the idea of calling how many different microphysics subroutines to align the cloud sizes with a microphysics scheme is going to get really nasty.  This is part of the larger picture of how future codes have to consider a plethora of connectivities  -none of which are trivial to resolve.  One could argue we need to call separate routines for NSSL, P3, Milbrandt, Goddard, Thompson, the list is crazy long and my own thought is this enhancement is way better than existing code and quite simple and isolated using the functions I built.

Reply:
Greg,
Do moist correctly. Pass it in as just "moist".  After you do your PR, we'll clean up scalar with a second PR. Also, if moist is an intent(in) variable only (which appears to be the case), please modify the declaration.